src/dbnode/integration/disk_coldflush_test.go

```diff
@@ -0,0 +1,217 @@
+			SetRetentionPeriod(12 * time.Hour)).
+		SetColdWritesEnabled(true)
+	nsID := ident.StringID("testColdWriteNs1")
+	ns, err := namespace.NewMetadata(nsID, nsOpts)
+	require.NoError(t, err)
+	testOpts := newTestOptions(t).
+		SetTickMinimumInterval(time.Second).
+		SetNamespaces([]namespace.Metadata{ns})
+
+	testSetup, err := newTestSetup(t, testOpts, nil)
+
+	require.NoError(t, err)
+	defer testSetup.close()
+
+	md := testSetup.namespaceMetadataOrFail(nsID)
+	ropts := md.Options().RetentionOptions()
+	blockSize := ropts.BlockSize()
+	filePathPrefix := testSetup.storageOpts.CommitLogOptions().FilesystemOptions().FilePathPrefix()
+
+	// Start the server.
+	log := testSetup.storageOpts.InstrumentOptions().Logger()
+	log.Debug("disk coldflush test")
+	require.NoError(t, testSetup.startServer())
+	log.Debug("server is now up")
+
+	// Stop the server.
+	defer func() {
+		require.NoError(t, testSetup.stopServer())
+		log.Debug("server is now down")
+	}()
+
+	// Write warm data first so that cold data will flush.
+	start := testSetup.getNowFn()
+	seriesMaps := make(map[xtime.UnixNano]generate.SeriesBlock)
+	warmData := []generate.BlockConfig{
+		{IDs: []string{"warm1", "warm2"}, NumPoints: 100, Start: start},
+		{IDs: []string{"warm1", "warm3"}, NumPoints: 50, Start: start.Add(blockSize)},
+	}
+
+	expectedDataFiles := []fs.FileSetFileIdentifier{
+		fs.FileSetFileIdentifier{
+			// warm1, start
+			Namespace:   nsID,
+			Shard:       6,
+			BlockStart:  start,
+			VolumeIndex: 0,
+		},
+		fs.FileSetFileIdentifier{
+			// warm1, start + 1
+			Namespace:   nsID,
+			Shard:       6,
+			BlockStart:  start.Add(blockSize),
+			VolumeIndex: 0,
+		},
+		fs.FileSetFileIdentifier{
+			// warm2, start
+			Namespace:   nsID,
+			Shard:       11,
+			BlockStart:  start,
+			VolumeIndex: 0,
+		},
+		fs.FileSetFileIdentifier{
+			// warm3, start + 1
+			Namespace:   nsID,
+			Shard:       2,
+			BlockStart:  start.Add(blockSize),
+			VolumeIndex: 0,
+		},
+	}
+	for _, input := range warmData {
+		testSetup.setNowFn(input.Start)
+		testData := generate.Block(input)
+		seriesMaps[xtime.ToUnixNano(input.Start)] = testData
+		require.NoError(t, testSetup.writeBatch(nsID, testData))
+	}
+	log.Debug("warm data is now written")
+
+	// Advance time to make sure all data are flushed. Because data
+	// are flushed to disk asynchronously, need to poll to check
+	// when data are written.
+	testSetup.setNowFn(testSetup.getNowFn().Add(blockSize * 2))
+	maxWaitTime := time.Minute
+	require.NoError(t, waitUntilFileSetFilesFlushed(filePathPrefix, expectedDataFiles, maxWaitTime))
+
+	// Verify on-disk data match what we expect.
+	verifyFlushedDataFiles(t, testSetup.shardSet, testSetup.storageOpts, nsID, seriesMaps)
+
+	coldData := []generate.BlockConfig{
+		{IDs: []string{"cold0"}, NumPoints: 80, Start: start.Add(-blockSize)},
+		{IDs: []string{"cold1", "cold2", "cold3"}, NumPoints: 30, Start: start},
+		{IDs: []string{"cold1", "cold3"}, NumPoints: 20, Start: start.Add(blockSize)},
```

Comment:
Want to do 2*blocksize for good measure (instead of just single block size?)

Reply:
I'll set `now` to `start + 3 * blockSize` to ensure these are cold writes.

Reply:
Sounds good.